catkin_ws/src/kpr_testing/src/valves.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+
+import rospy, termios, sys, contextlib, os
+import numpy as np
+from ur_msgs.srv import *
+from ur_msgs.msg import *
+
+header = """
+Check the status of I/O-pins
+\t-^D to exit
+\t-i to change income settings
+\t-p to send io states
+\t-t to toggle service succes
+"""
+
+def display():
+	os.system('clear')
+	print header
+#Making table
+	tableHeader = "Pin     |  State     Fun     "
+	print tableHeader
+	print "-"*len(tableHeader)
+	print "VacSens | ", info[VacSens][0], "  ", info[VacSens][1]
+	print "Gripper | ", info[Gripper][0], "  ", info[Gripper][1]
+
+#These IO pins are used
+VacSens = "DI1"
+Gripper = "DO2"
+
+#Set succes response
+response = True
+
+#dictionary of pins
+info = {VacSens:["-","     -"], Gripper:["-","     -"]}
+
+#needed for the "pressing keys"
+@contextlib.contextmanager
+def raw_mode(file):
+    old_attrs = termios.tcgetattr(file.fileno())
+    new_attrs = old_attrs[:]
+    new_attrs[3] = new_attrs[3] & ~(termios.ECHO | termios.ICANON)
+    try:
+        termios.tcsetattr(file.fileno(), termios.TCSADRAIN, new_attrs)
+        yield
+    finally:
+        termios.tcsetattr(file.fileno(), termios.TCSADRAIN, old_attrs)
+
+def request(req):
+	global response
+	display()
+#Checking if changes wil be made to used pins 
+	if req.fun == 1:
+		pin = 'DO' + str(req.pin)
+	elif req.fun == 4:
+		pin = 'AO' + str(req.pin)
+	else:
+		rospy.logerr("Writing to unused pin")
```

Comment:
Correct me if i'm wrong, but shouldn't this message say "You cannot write to an input pin"?